app/src/main/java/fr/free/nrw/commons/nearby/NearbyActivity.java

```diff
@@ -198,23 +222,29 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         }
     }
 
-    private void showMapView() {
+    private void toggleView() {
```

Comment:
If you added a 'toggle()' method on the enum, then the complexity of this method actually falls away
```java
if (nearbyAsyncTask != null) {
    viewMode = viewMode.toggle();
    if (nearbyAsyncTask.getStatus() == AsyncTask.Status.FINISHED) {
        if (viewMode.isMap()) {
            setMapFragment();
        } else {
            setListFragment();
        }
    }
    sharedPreferences.edit().putBoolean(MAP_LAST_USED_PREFERENCE, viewMode.isMap()).apply();
}
```